pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+            property_list = [p.lower() for p in property_list]
+            for pname in obj.properties.keys():
+                if pname.lower() not in property_list:
+                    del obj.properties[pname]
+
+    ###############################################################
+    #
+    #   Access  to data store
+    #
+    #   The following methods provide access to the data store for
+    #   a single namespace. There are separate methods to provide
+    #   access to each of the 3 object stores within the data store
+    #   CIMClass, CIMInstance, and CIMQualifierDeclaration.
+    #
+    ##################################################################
+
+    def get_class_store(self, namespace):
+        """
+        Returns the class data store for the specified CIM namespace within the
+        data store.  This method validates that the namespace exists in the
+        data store.
+
+        If the class object store does not contain the namespace an exception
+        is generated.
+
+        Parameters:
+
+          namespace(:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing
+            slash charactes are ignored.
+
+        Returns:
+
+          Instance of :class:`~pywbem_mock.`BaseObjectStore` which is the
+          object store for classes in the CIM repository.
+
+        Raises:
+
+          :exc:`~pywbem.CIMError`: CIM_ERR_INVALID_NAMESPACE: Namespace does
+            not exist.
+        """
+        self.validate_namespace(namespace)
+        return self.cimrepository.get_class_store(namespace)
+
+    def get_instance_store(self, namespace):
+        """
+        Returns the instance object store for the specified CIM namespace
+        within the CIM repository.  This method validates that the namespace
+        exists.
+
+        If the object store does not contain the namespace an exception
+        is generated.
+
+        Parameters:
+
+          namespace(:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing
+            slash charactes are ignored.
+
+        Returns:
+
+          :class:`~pywbem_mock.`BaseObjectStore` which is the object store for
+           instances in the CIM repository.
+
+        Raises:
+
+          :exc:`~pywbem.CIMError`: CIM_ERR_INVALID_NAMESPACE: Namespace does
+            not exist.
+        """
+
+        self.validate_namespace(namespace)
+        return self.cimrepository.get_instance_store(namespace)
+
+    def get_qualifier_store(self, namespace):
+        """
+        Returns the qualifier declaration object store for the specified CIM
+        namespace within the CIM repository.  This method validates that the
+        namespace exists.
+
+        If the object store does not contain the namespace an exception
+        is generated.
```

Comment:
Along the reasoning in a previous comment, change "object store" to "CIM repository".

Reply:
DONE